src/include/libpmemobj.h

```diff
@@ -942,13 +942,17 @@ pmemobj_tx_free((o).oid)
 
 #define	TX_MEMCPY(dest, src, num) (\
 {\
-	pmemobj_tx_add_range_direct(dest, num);\
-	memcpy(dest, src, num); })
+	void *d = (dest);\
+	size_t n = (num);\
```

Comment:
Declaring temporary variables in macros can have unexpected consequences when the name collides with existing variables.  In this case, it is not hard to imagine someone using `d` for their dest variable and `n` for their count, and the declarations here would not then do what is expected (and no compile-time check would find the error).

If you feel you must use temporary variables declared in macros, I suggest using the convention of prepending underscores to make a name collision less likely.  But honestly inline functions pretty much remove the need for these games.  Why not just change TX_MEMCPY() to an inline function?